app/os_macos.m

```diff
@@ -110,6 +110,27 @@ - (void)mouseMoved:(NSEvent *)event {
 - (void)mouseDragged:(NSEvent *)event {
 	handleMouse(self, event, MOUSE_MOVE, 0, 0);
 }
+-(NSDragOperation)draggingEntered:(id < NSDraggingInfo >)sender
+{
+    return NSDragOperationCopy; // NSDragOperationGeneric
+}
+- (void)draggingEnded:(id <NSDraggingInfo>)sender
+{
+	NSPasteboard* pbrd = [sender draggingPasteboard];
+
+	for (NSPasteboardItem *item in [pbrd pasteboardItems]) {
+		for (NSString *type in [item types]) {
+			if ([type isEqualToString:@"public.file-url"]) {
+				NSData* data = [item dataForType:type];
+				NSString *clipboard_url = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+				NSURL *url = [[NSURL alloc] initWithString:clipboard_url];
+				NSString *real_path = url.path;
+				char* content = (char*)[real_path UTF8String];
```

Comment:
I suggest replacing this with passing the `NSData *data` to Go and let Go code parse it as an URL. I believe `NSData` has a [`bytes`](https://developer.apple.com/documentation/foundation/nsdata/1410616-bytes) property to get a pointer to the data, along with a `length` for its length.